File: OurTest/cameraTest.cpp
#include <opencv2/opencv.hpp>
#include <iostream>

using namespace cv;
using namespace std;

int main() {
    // URL of the video stream from your mobile phone camera
    string url = "http://192.168.1.125:8080/video";

    // Open the video stream
    VideoCapture cap(url);

    if (!cap.isOpened()) {
        cerr << "Error opening video stream or file" << endl;
        return -1;
    }

    // Set properties to reduce resolution and FPS
    cap.set(CAP_PROP_FRAME_WIDTH,320);
    cap.set(CAP_PROP_FRAME_HEIGHT, 200);
    cap.set(CAP_PROP_FPS, 1);

    while (true) {
        Mat frame;
        // Capture frame-by-frame
        cap >> frame;

        // If the frame is empty, break immediately
        if (frame.empty())
            break;

        // Operations on the frame (if needed)
        Mat gray;
        cvtColor(frame, gray, COLOR_BGR2GRAY);

        // Display the frame
        imshow("Frame", gray);

        // Exit if 'q' is pressed
        if (waitKey(1) == 'q')
            break;
    }

    // Release the VideoCapture object
    cap.release();
    destroyAllWindows();

    return 0;
}

File: OurTest/cameraTest.py
import cv2
import numpy as np

url = 'http://192.168.1.125:8080/video'


cap = cv2.VideoCapture(url) # check this
while(True):
    # Capture frame-by-frame
    ret, frame = cap.read()
    # Our operations on the frame come here
    gray = cv2.cvtColor(frame, cv2.COLOR_BGR2GRAY)
    # Display the resulting frame
    cv2.imshow("frame",gray)
    if cv2.waitKey(1) & 0xFF == ord("q"):
        break
# When everything done, release the capture
cap.release()